CMake with error 'No target "daslang"' when DAS_TOOLS_DISABLED is ON

Wanted to only link libDaScript and disable everything else which was working... but after syncing a while back this error started showing up when I try reloading cmake
```
CMake Error at .../daScript/CMakeLists.txt:543 (ADD_CUSTOM_TARGET):
  Error evaluating generator expression:

    $<TARGET_FILE:daslang>

  No target "daslang"
```

which was coming from the if block `IF(DAS_EXAMPLES_TO_RUN)` at line '499'

this happens because it seems the target "daslang" is only added if 'DAS_TOOLS_DISABLED' was 'OFF' (which I had set to 'ON')

This happened like around a month back or so, thought it could be a temp thing but I just pulled and I was getting the same error (I fixed it temporarily by just doing `IF(DAS_EXAMPLES_TO_RUN AND FALSE)`)
Thought maybe I should open an issue if you weren't aware of it

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1755,10 +1755,13 @@ install(FILES
 add_subdirectory(tutorials)
 
 # ── Documentation build (optional) ──────────────────────────────────────────
-# Requires the `daslang` target (das2rst stdlib RST generator runs it), so we
-# also gate on NOT DAS_TOOLS_DISABLED — otherwise CMake fails on the missing
-# target. Tools-disabled builds simply can't generate docs.
-if(DAS_BUILD_DOCUMENTATION AND NOT DAS_TOOLS_DISABLED)
+# Requires the `daslang` target (das2rst stdlib RST generator runs it). If the
+# user explicitly asked for docs but also disabled tools, fail early rather
+# than silently skipping — mirrors the DAS_AOT_EXAMPLES check above.
+if(DAS_BUILD_DOCUMENTATION)
+    if(DAS_TOOLS_DISABLED)
+        message(FATAL_ERROR "DAS_BUILD_DOCUMENTATION requires DAS_TOOLS_DISABLED to be OFF (the docs pipeline runs daslang via das2rst.das)")
+    endif()
     find_package(Python3 COMPONENTS Interpreter)
     if(Python3_FOUND)
         execute_process(

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -438,13 +438,22 @@ ENDMACRO()
 # Register a .das file to be run as a headless example check.
 # Modules call this from their CMakeLists.txt for examples that
 # can run without GUI, audio, or user interaction.
+#
+# When DAS_TOOLS_DISABLED=ON the `daslang` target is never created, so the
+# run_examples target below (which references $<TARGET_FILE:daslang>) cannot
+# be generated. Gate the registration here so every call site is automatically
+# safe and DAS_EXAMPLES_TO_RUN stays empty.
 MACRO(ADD_EXAMPLE_RUN das_file)
-    SET(_exe TRUE)
-    IF("${ARGN}" STREQUAL "FALSE")
-        SET(_exe FALSE)
+    IF(DAS_TOOLS_DISABLED)
+        MESSAGE(STATUS "EXAMPLE_RUN ${das_file} skipped (DAS_TOOLS_DISABLED=ON)")
+    ELSE()
+        SET(_exe TRUE)
+        IF("${ARGN}" STREQUAL "FALSE")
+            SET(_exe FALSE)
+        ENDIF()
+        MESSAGE("EXAMPLE_RUN ${das_file} (exe=${_exe})")
+        LIST(APPEND DAS_EXAMPLES_TO_RUN "${das_file}|${_exe}")
     ENDIF()
-    MESSAGE("EXAMPLE_RUN ${das_file} (exe=${_exe})")
-    LIST(APPEND DAS_EXAMPLES_TO_RUN "${das_file}|${_exe}")
 ENDMACRO()
 
 ADD_EXAMPLE_RUN("${PROJECT_SOURCE_DIR}/examples/dasbind/dasbind_example.das")
@@ -1117,7 +1126,11 @@ SET(DAS_DASCRIPT_MAIN_SRC
 
 # Tests
 if (NOT ${DAS_TESTS_DISABLED})
-    include(examples/pathTracer/CMakeLists.txt)
+    # pathTracer links against libDaScriptAot, which is only created when
+    # AOT examples are enabled — gate the inclusion accordingly.
+    if(NOT ${DAS_AOT_EXAMPLES_DISABLED})
+        include(examples/pathTracer/CMakeLists.txt)
+    endif()
     include(examples/crash/CMakeLists.txt)
     if(NOT ${DAS_AOT_EXAMPLES_DISABLED} AND NOT (WIN32 AND CMAKE_SIZEOF_VOID_P EQUAL 4))
         include(tests/aot/CMakeLists.txt)
@@ -1742,7 +1755,10 @@ install(FILES
 add_subdirectory(tutorials)
 
 # ── Documentation build (optional) ──────────────────────────────────────────
-if(DAS_BUILD_DOCUMENTATION)
+# Requires the `daslang` target (das2rst stdlib RST generator runs it), so we
+# also gate on NOT DAS_TOOLS_DISABLED — otherwise CMake fails on the missing
+# target. Tools-disabled builds simply can't generate docs.
+if(DAS_BUILD_DOCUMENTATION AND NOT DAS_TOOLS_DISABLED)
     find_package(Python3 COMPONENTS Interpreter)
     if(Python3_FOUND)
         execute_process(

